compiler-rs/clients_schema_to_openapi/src/schemas.rs

```diff
@@ -468,15 +465,171 @@ impl<'a> TypesAndComponents<'a> {
+    };
+
+    if io.typ.is_builtin() {
+        return Ok(None);
+    }
+
+    let TypeDefinition::TypeAlias(alias) = model.get_type(&io.typ)? else {
+        return Ok(None);
+    };
+
+    // Try to unwrap further or else return the current alias
+    let result = match unwrap_alias(&alias.typ, model)? {
+        Some(alias_value) => Some(alias_value),
+        None => Some(&alias.typ),
+    };
+
+    Ok(result)
+}
+
+/// Checks if a value_of is a lenient array definition (i.e. `Foo | Foo[]`) and
+/// if successful, returns the value definition.
+fn unwrap_lenient_array(value: &ValueOf) -> Option<&ValueOf> {
+    // Is this a union
+    let ValueOf::UnionOf(u) = value else {
+        return None
+    };
+
+    // of a value and array_of (in any order)
+    let (single_value, array_value) = match &u.items.as_slice() {
+        [v, ValueOf::ArrayOf(ao)] |
+        [ValueOf::ArrayOf(ao), v] => (v, &*ao.value),
+        _ => return None,
+    };
+
+    // and both value types are the same
+    if single_value == array_value {
+        return Some(single_value);
+    }
+
+    None
+}
+
+fn unwrap_array(value: &ValueOf) -> Option<&ValueOf> {
+    match value {
+        ValueOf::ArrayOf(ArrayOf { value }) => Some(value),
+        _ => None,
+    }
+}
+
+/// If a property value is an enumeration (possibly via aliases and arrays)
+fn lift_enum_descriptions(prop: &Property, model: &clients_schema::IndexedModel) -> anyhow::Result<Option<String>> {
+
+    // FIXME: could be memoized on `prop.typ` as we'll redo this work every time we encounter the same value definition
+    let value = &prop.typ;
+
+    // Maybe an alias pointing to an array or lenient array
+    let value = unwrap_alias(value, model)?.unwrap_or(value);
+
+    // Unwrap lenient array
+    let (lenient_array, value) = match unwrap_lenient_array(value) {
+        Some(lenient_array) => (true, lenient_array),
+        None => (false, value),
+    };
+
+    // Unwrap array to get to the enum type
+    let value = unwrap_array(value).unwrap_or(value);
+
+    // Unwrap aliases again, in case the array value was itself an alias
+    let value = unwrap_alias(value, model)?.unwrap_or(value);
+
+    // Is this an enum?
+    let ValueOf::InstanceOf(inst) = value else {
+        return Ok(None);
+    };
+
+    if inst.typ.is_builtin() {
+        return Ok(None);
     }
+
+    let TypeDefinition::Enum(enum_def) = model.get_type(&inst.typ)? else {
+        return Ok(None);
+    };
+
+    let mut result: String = match &prop.description {
+        Some(desc) => desc.clone(),
+        None => String::new(),
+    };
+
+    // Do we have at least one enum member description?
+    if enum_def.members.iter().any(|m| m.description.is_some()) {
+        // Some descriptions: output a list with descriptions
+
+        // Close description paragraph and add an empty line to start a new paragraph
+        writeln!(result)?;
+        writeln!(result)?;
+
+        writeln!(result, "Values are:")?;
```

Comment:
To avoid confusion around the duplication of this text with the automated text, can we change it to something like this?:
```suggestion
        writeln!(result, "Supported values include:")?;
```